### DISCUSSION TITLE
SD2.1 Base / Dreambooth training with extension

### DISCUSSION THREAD

**Discussion by mediocreatmybest** [d8ahazard/sd_dreambooth_extension#586] [Q&A]
Hey! I've found when creating a checkpoint from the interface, if the model is the SD2.x 512x the yaml file that is created assumes the SD2.x was the 768-V version. 

As a consequence if the SD2.x was the base version without V-Prediction the model won't render correctly and just generates noise. When replacing it with the correct yaml file it starts to render correctly, I guess my question is: when training is the correct settings being used? Or should only the 768-V be the only one trained? 

Thanks!

**Comment by mediocreatmybest** [comment]
> Hey! I've found when creating a checkpoint from the interface, if the model is the SD2.x 512x the yaml file that is created assumes the SD2.x was the 768-V version. 
> 
> As a consequence if the SD2.x was the base version without V-Prediction the model won't render correctly and just generates noise. When replacing it with the correct yaml file it starts to render correctly, I guess my question is: when training is the correct settings being used? Or should only the 768-V be the only one trained? 
> 
> Thanks!


Just checked the json file, yeah looks like it's flagged as v_prediction after it builds the training image from the source checkpoint. So I'm guessing it doesn't pull information from the yaml file that's included with the source model?

**Comment by nanafy** [comment]
I've seen this as well. the 512 version gets assumed to be the same as 768, good catch!

**Comment by zahdab** [comment]
I've noticed that too when i was facing issues generating class images https://github.com/d8ahazard/sd_dreambooth_extension/discussions/531,
what i've been doing since then is that you can train the 768 model using 512 images and retain the benefits of 768 ... u just need to feed 512 images and change the parameter in the training to reflect that. 

Also i would advise not exceed 0.0000015 learning rates and using as few images as possible ... latest trick for me for character training is to train the closeup photos at a 0.0000015 learning rate and the further away photo as a 0.0000012 learning rate.  100 steps per image

**Reply by mediocreatmybest** [reply]
Is that with the current version? I'm reinstalling my Linux box tonight and reloading all the training / programs. I've only been able to use google colab so far and haven't been successful in getting SD2.1 training at all. Even with the correct yaml file. 

Hopefully tonight I'll be able to test that out with a native Linux box that isn't colab.

**Reply by zahdab** [reply]
sorry i update the comment, i butchered the learning rates, i was about the to sleep.

**Comment by yanfeiwong** [comment]
And somehow seems like you need to creat a V2.1 512 checkpoint from the interface, to download some stuff then you can creat a V2.1 768 checkpoint...

**Comment by TWIISTED-STUDIOS** [comment]
how are you training with the 2.1 model everytime I try the classification images just come out as blank

Edited** Never mind fixed it needed to select xformers as memory allocation as if left blank they came up as black squares

**Reply by yanfeiwong** [reply]
I do not use classification and xformers for now... training both text encoder and ema, 2it/s on a 3090, 18~22g vram...

**Reply by zahdab** [reply]
set memory attention to xformers, i had the same issue with class images and that's how i resolved it.
